tms/models/res_partner.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+DRIVER_LICENSE_TYPES = [
+    ("A", "A - Motorcycles"),
+    ("B", "B - Automobiles"),
+    ("C", "C - Truck"),
+    ("D", "D - Bus"),
+]
+LOCATION_TYPES = [("terrestrial", "Terrestrial")]
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+    # TMS Type
+    tms_type = fields.Selection(
+        string="Type",
+        selection=[("driver", "Driver"), ("location", "Location")],
+        default=lambda self: self.env.context.get("default_tms_type"),
+    )
+
+    # ------------------------------
+    #            Driver
+    # ------------------------------
+
+    # Driver - Flags
+    is_external = fields.Boolean(string="External Driver")
+    is_training = fields.Boolean(string="In Training")
+    is_active = fields.Boolean(default=True)
+
+    # Driver - Information
+    tms_age = fields.Integer()  # TODO: Add min age
+    gender = fields.Selection(selection=[("male", "Male"), ("female", "Female")])
+
+    # Driver - Relations
+    vehicles_ids = fields.One2many("fleet.vehicle", "driver_id")
+    trips_ids = fields.One2many("tms.order", "driver_id")
+
+    # Driver - Type
+    driver_type = fields.Selection(
+        string="Type", selection=[("terrestrial", "Terrestrial")]
+    )
+
+    # Driver - Relations
+    team_id = fields.Many2one("tms.team")
+    crew_ids = fields.Many2many(
+        "tms.crew",
+        "tms_crew_drivers_rel",
+        string="Crews",
+    )
+    stage_id = fields.Many2one(
```

Comment:
Knowing that the stage can be registered for a company, I believe this field should depend on the company.